signer/tuf_on_ci_sign/delegate.py

```diff
@@ -89,11 +92,35 @@ def verify_signers(response: str) -> list[str]:
         if choice == 0:
             break
         if choice == 1:
-            config.signers = click.prompt(
-                bold(f"Please enter list of {role} signers"),
-                default=", ".join(config.signers),
-                value_proc=verify_signers,
-            )
+            # if role is not root or targets, allow online keys
+            if role in ["root", "targets"]:
+                config.signers = click.prompt(
+                    bold(f"Please enter list of {role} signers"),
+                    default=", ".join(config.signers),
+                    value_proc=verify_signers,
+                )
+            else:
+                click.echo("Choose what keytype to use:")
+                click.echo("1. Configure offline signers:")
+                click.echo("2. Configure online signers")
+                signer_choice = click.prompt(
+                    bold("Please choose an option or press enter to continue"),
+                    type=click.IntRange(1, 2),
+                )
+
+                if signer_choice == 1:
+                    config.signers = click.prompt(
+                        bold(f"Please enter list of {role} signers"),
+                        default=", ".join(config.signers),
+                        value_proc=verify_signers,
+                    )
+                elif signer_choice == 2:
+                    toplevel = git_expect(["rev-parse", "--show-toplevel"])
+                    settings_path = os.path.join(toplevel, ".tuf-on-ci-sign.ini")
+                    user_config = User(settings_path)
+                    online_key = _collect_online_key(user_config)
+                    uri = online_key.unrecognized_fields[TAG_ONLINE_URI]
+                    config.signers = [uri]
```

Comment:
I left some higher level comments on this in the issue: I'm not sure the online keys should be a special case -- why not consider them just one of the signers you can add?

Reply:
I guess my suggestion would require some re-design of adding signers since the KMS keys need more info so maybe not worth it

Reply:
That was the reason I made them special, the code is easier. And also I think the flow for the repository maintainers is simplified if they just provide all the details for the online key direct.